### PR TITLE
Enable pipes on stdin and stdout

### PR DESCRIPTION
**Pull request by Ozzard**
In theory, this allows - for stdin or stdout on -i and -o options. In reality, even the most recent version of the "getopt-compatible" command line parser we use is not fully compatible; I suggest we merge this patch now and wait for a new release of the command line parser, at which point this should suddenly Just Work.

### CODE REVIEW

File: MzmlParser/MzmlReader.cs [hunk 1]
```diff
@@ -110,7 +110,7 @@ public Run LoadMzml(string path, AnalysisSettings analysisSettings)
         private void ReadMzml(string path, Run run, bool irt)
         {
 
-            using (XmlReader reader = XmlReader.Create(path))
+            using (XmlReader reader = "-".Equals(path) ? XmlReader.Create(Console.In): XmlReader.Create(path))
```
Comment (PaulBrack) [inline MzmlParser/MzmlReader.cs:113]: Can we put in an explantory comment here?

File: MzqcGenerator/MzqcWriter.cs [hunk 2]
```diff
@@ -127,7 +128,7 @@ public void BuildMzqcAndWrite(string outputFileName, Run run, Dictionary<string,
 
         public void WriteMzqc(string path, JsonClasses.MzQC metrics)
         {
-            using (StreamWriter file = File.CreateText(path))
+            using (TextWriter file = "-".Equals(path, StringComparison.Ordinal) ? Console.Out : File.CreateText(path))
```
Comment (PaulBrack) [inline MzqcGenerator/MzqcWriter.cs:131]: Explanatory comment please

File: MzmlParser/MzmlReader.cs (hunk 1 above)
Comment (PaulBrack) [reply MzmlParser/MzmlReader.cs:113]: No string culture on this one but there is on the other one?

File: SwaMe.Console/Program.cs [hunk 3]
```diff
@@ -44,7 +44,9 @@ static void Main(string[] args)
                     MaxThreads = options.MaxThreads
                 };
 
-                CheckFileIsReadableOrComplain(options.InputFile);
+                // stdin is always considered readable; anything else needs a check (#99)
+                if (!"-".Equals(options.InputFile))
```
Comment (PaulBrack) [inline SwaMe.Console/Program.cs:48]: Magic string shared across 3 projects - very small number of locations and unlikely to change, but would prefer it if we made this evaluation in the calling assembly and passed a StdIn / StdOut bool on the read/write method call

Comment (Ozzard) [reply SwaMe.Console/Program.cs:48]: Agree about parsing "-" in a console app.  An alternative would be to pass null to indicate "no file, use input/output"; thoughts?

Comment (Ozzard) [reply SwaMe.Console/Program.cs]: Moved to using nulls.

File: MzqcGenerator/MzqcWriter.cs (hunk 2 above)
Comment (Ozzard) [reply MzqcGenerator/MzqcWriter.cs]: Added as comment to parameter.

File: MzmlParser/MzmlReader.cs (hunk 1 above)
Comment (Ozzard) [reply MzmlParser/MzmlReader.cs]: Added explanatory comments and correct StringComparisons.

**Pull request merged**